File: src/main/golang/leetcode/editor/cn/[204]Count Primes.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func countPrimes(n int) (ans int) {
	var notPrime = make([]bool, n)
	isPrime := func(x int) bool {
		for i := 2; i*i < x; i++ {
			if x%i == 0 {
				return false
			}
		}
		return true
	}
	for i := 2; i*i < n; i++ {
		if isPrime(i) {
			for j := i * i; j < n; j += i {
				notPrime[j] = true
			}
		}
	}
	for i := 2; i < n; i++ {
		if !notPrime[i] {
			ans += 1
		}
	}
	return
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[204]Count Primes.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 3:
            return 0
        is_prime = [True] * n
        is_prime[0] = is_prime[1] = False
        for i in range(2, int(n ** 0.5) + 1):
            if is_prime[i]:
                is_prime[i * i:n:i] = [False] * len(is_prime[i * i:n:i])
        return sum(is_prime)
# leetcode submit region end(Prohibit modification and deletion)
